### PR TITLE
additional tests

### PR DESCRIPTION
**Pull request by erral**
As a follow-up of #1836, I add additional tests showing that we return an error when trying to change a login to an existing one (see https://github.com/plone/plone.restapi/pull/1836#pullrequestreview-2424965276)

----
📚 Documentation preview 📚: https://plonerestapi--1840.org.readthedocs.build/

### CODE REVIEW

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/test_services_users.py [hunk 1]
```diff
@@ -1663,3 +1663,128 @@ def test_user_changes_email_when_login_with_email_and_uuid_userids(self):
...
+        self.assertTrue(anon_response.ok)
+
+        # try to change the email to an existing one, it should fail
+        email_change_response = self.api_session.patch(
+            f"/@users/{userid}",
+            json={
+                "email": "second@example.com",
+            },
+        )
+        self.assertFalse(email_change_response.ok)
```
Comment (davisagli) [inline src/plone/restapi/tests/test_services_users.py:1716]: I can't tell from this test whether the error handling is correct. What is the status code? What is the message?

Comment (erral) [reply src/plone/restapi/tests/test_services_users.py:1716]: You're right, sorry, I was testing that at least Plone returns an error and not testing a meaningful error.

I have found that I was returning a 500 error when a 400 one should be more appropriate. I have amended the code and the tests.

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

**Review by davisagli — APPROVED** [review]

**Pull request merged**